packages/core/core-flows/src/cart/workflows/update-cart-promotions.ts

```diff
@@ -42,10 +43,18 @@ export const updateCartPromotionsWorkflow = createWorkflow(
       list: false,
     })
 
+    const promo_codes = transform({ input }, (data) => {
+      return (data.input.promo_codes || []) as string[]
+    })
+
+    const action = transform({ input }, (data) => {
+      return data.input.action || PromotionActions.ADD
+    })
+
     const promotionCodesToApply = getPromotionCodesToApply({
       cart: cart,
-      promo_codes: input.promo_codes ?? [],
-      action: input.action || PromotionActions.ADD,
+      promo_codes,
```

Comment:
on another note, this could probably also be handled by the step itself, so that it is more resuable

Reply:
Yes 100% but it is also handled I believe